service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderController.java

```diff
@@ -51,8 +53,28 @@ public AboutToStartOrSubmitCallbackResponse handleMidEventValidateAddress(
             .build();
     }
 
-    @PostMapping("/date/mid-event")
-    public AboutToStartOrSubmitCallbackResponse handlleMidEventValidateDate(
+    @PostMapping("/interim-end-date/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEventValidateInterimEndDate(
+        @RequestBody CallbackRequest callbackRequest) {
+
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+        final InterimEndDate interimEndDate = caseData.getInterimEndDate();
+
+        List<String> errors = List.of();
+
+        if (interimEndDate.getType() == NAMED_DATE) {
+            errors = validateGroupService.validateGroup(interimEndDate, Default.class);
```

Comment:
should we be validating `intermEndDate` using the default class? Might be safer grouping.

Reply:
Added a group